assets/js/payment-method-extensions/payment-methods/stripe/credit-card/payment-method.js

```diff
@@ -34,9 +34,9 @@ const CreditCardComponent = ( {
 	const { customerId } = billing;
 	const [ sourceId, setSourceId ] = useState( '' );
 	const stripe = useStripe();
-	const [ shouldSavePayment, setShouldSavePayment ] = useState(
-		customerId ? true : false
-	);
+	// Saving payment method is always opt-in (based on previous/shortcode
+	// behaviour); so default to false.
+	const [ shouldSavePayment, setShouldSavePayment ] = useState( false );
```

Comment:
The design called for checked by default, but I do think this should be opt-in so I'm fine with this change. 

cc @garymurray just for a gut check here. Gary, I pinged you to get your feedback on whether the "Save payment information to my account for future purchase." checkbox should default to checked or default to unchecked. Currently (according to Rua) the shortcode checkout has this checkbox default to _unchecked_.  Original implementation prior to this pull was to default to checked in keeping with the design specs but its unclear whether the design was communicating the desired default or not.